### DISCUSSION TITLE
Low-level errors during `Issuer.discover()` cause error without stack trace

### DISCUSSION THREAD

**Discussion by sybereal** [panva/openid-client#598] [Ideas]
**Describe the bug**

Below-HTTP errors, e.g. connection reset, during `Issuer.discover()` throw errors without a useful stack trace. Example:

```
Error: read ECONNRESET
    at __node_internal_captureLargerStackTrace (node:internal/errors:478:5)
    at __node_internal_errnoException (node:internal/errors:608:12)
    at TLSWrap.onStreamRead (node:internal/stream_base_commons:217:20) {
  errno: -104,
  code: 'ECONNRESET',
  syscall: 'read'
}
```

**To Reproduce**
Issuer and Client configuration: N/A - Crash occurred before these could be constructed.

Steps to reproduce the behaviour:

1.  `await Issuer.discover("https://some-host-that-resets.com")

**Expected behaviour**
The stack trace should include the call to `Issuer.discover()` and everything leading up to that. This is the case when the request is successful but a HTTP error, such as 404, is returned:

```
> await Issuer.discover('https://example.com')
Uncaught OPError: expected 200 OK, got: 404 Not Found
    at processResponse (/app/node_modules/openid-client/lib/helpers/process_response.js:41:11)
    at Issuer.discover (/app/node_modules/openid-client/lib/issuer.js:179:18)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async REPL2:1:33 {
  error: 'expected 200 OK, got: 404 Not Found'
}
```

**Environment:**
 - openid-client version: 5.4.0
 - node version: v16..19.0

**Additional context**

 - [x] the bug is happening on latest openid-client too.
 - [x] i have searched the issues tracker on github for similar issues and couldn't find anything related.

**Comment by panva** [comment]
I wouldn't mind if the calls to `lib/helpers/request.js` had a rejection handler with a rich message RPError and the underlying network errors as [cause](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause).

Would you like to open a PR to get this started?
